Update FreeBSD version in Cirrus CI configuration

pheiduck opened the pull request:
Image names are update:
- https://cirrus-ci.org/guide/FreeBSD/#list-of-available-image-families

Freebsd 14.3 Ends in 6 months and 3 weeks
(30 Jun 2026)

pheiduck committed:
Update FreeBSD version in Cirrus CI configuration

Image names are update:
- https://cirrus-ci.org/guide/FreeBSD/#list-of-available-image-families

pheiduck committed:
fixup: add vterm support to configure script

pheiduck committed:
fixup: set TERM environment variable for test script

chrisbra commented:
I am slightly suprised about the `export TERM=xterm` Is this really needed? Also the `--with-vterm=builtin` is an unknown configure argument

pheiduck commented:
I added this because of this:

```if test -n "${ASAN_OPTIONS}"; then  ASAN_OPTIONS="${ASAN_OPTIONS}_opt_test" UBSAN_OPTIONS="${UBSAN_OPTIONS}_opt_test" ../vim -e -s -u NONE -U NONE --noplugin --not-a-term --nofork --gui-dialog-file guidialog -S util/gen_opt_test.vim ../optiondefs.h ../../runtime/doc/options.txt ;  else  ../vim -e -s -u NONE -U NONE --noplugin --not-a-term --nofork --gui-dialog-file guidialog -S util/gen_opt_test.vim ../optiondefs.h ../../runtime/doc/options.txt ;  fi
/* vi:set ts=8 sts=4 sw=4 noet:


From test_diffmode.vim:
Found errors in Test_diff_screen():
Run 1, 15:07:59 - 15:08:02 in  3.276525 seconds:
command line..script /tmp/cirrus-ci-build/src/testdir/runtest.vim[636]..function RunTheTest[63]..Test_diff_screen[36]..VerifyBoth[15]..VerifyScreenDump line 102: unified: See dump file difference: call term_dumpdiff("testdir/failed/Test_diff_01.dump", "testdir/dumps/Test_diff_01.dump"); difference in line 1: "| +0#0000e05#a8a8a8255@1|1+0#0000000#ffffff0| @33||+1&&| +0#0000e05#a8a8a8255@1|0+0#0000000#5fd7ff255| @33"; difference in line 2: "| +0#0000e05#a8a8a8255@1|-+0#4040ff13#afffff255@34||+1#0000000#ffffff0| +0#0000e05#a8a8a8255@1|1+0#0000000#ffffff0| @33"; difference in line 8: "| +0#0000e05#a8a8a8255@1|7+0#0000000#ffffff0| @33||+1&&|++0#0000e05#a8a8a8255| |+|-@1| @1|4| |l|i|n|e|s|:| |7|-@19"; difference in line 9: "|+| |+|-@1| @1|3| |l|i|n|e|s|:| |8|-@19||+1#0000000#ffffff0|~+0#4040ff13&| @35"
command line..script /tmp/cirrus-ci-build/src/testdir/runtest.vim[636]..function RunTheTest[63]..Test_diff_screen[40]..VerifyBoth[15]..VerifyScreenDump line 102: unified: See dump file difference: call term_dumpdiff("testdir/failed/Test_diff_02.dump", "testdir/dumps/Test_diff_02.dump"); difference in line 1: "| +0#0000e05#a8a8a8255@1|0+0#0000000#5fd7ff255| @33||+1&#ffffff0| +0#0000e05#a8a8a8255@1|1+0#0000000#ffffff0| @33"; difference in line 2: "| +0#0000e05#a8a8a8255@1|1+0#0000000#ffffff0| @33||+1&&| +0#0000e05#a8a8a8255@1|-+0#4040ff13#afffff255@34"; difference in line 8: "|++0#0000e05#a8a8a8255| |+|-@1| @1|4| |l|i|n|e|s|:| |7|-@19||+1#0000000#ffffff0| +0#0000e05#a8a8a8255@1|7+0#0000000#ffffff0| @33"; difference in line 9: "|~+0#4040ff13&| @35||+1#0000000&|++0#0000e05#a8a8a8255| |+|-@1| @1|3| |l|i|n|e|s|:| |8|-@19"; difference in line 10: "|~+0#4040ff13#ffffff0| @35||+1#0000000&|~+0#4040ff13&| @35"
Run 2, 15:08:04 - 15:08:07 in  3.361629 seconds:
command line..script /tmp/cirrus-ci-build/src/testdir/runtest.vim[677]..function RunTheTest[63]..Test_diff_screen[36]..VerifyBoth[15]..VerifyScreenDump line 102: unified: See dump file difference: call term_dumpdiff("testdir/failed/Test_diff_01.dump", "testdir/dumps/Test_diff_01.dump"); difference in line 1: "| +0#0000e05#a8a8a8255@1|1+0#0000000#ffffff0| @33||+1&&| +0#0000e05#a8a8a8255@1|0+0#0000000#5fd7ff255| @33"; difference in line 2: "| +0#0000e05#a8a8a8255@1|-+0#4040ff13#afffff255@34||+1#0000000#ffffff0| +0#0000e05#a8a8a8255@1|1+0#0000000#ffffff0| @33"; difference in line 8: "| +0#0000e05#a8a8a8255@1|7+0#0000000#ffffff0| @33||+1&&|++0#0000e05#a8a8a8255| |+|-@1| @1|4| |l|i|n|e|s|:| |7|-@19"; difference in line 9: "|+| |+|-@1| @1|3| |l|i|n|e|s|:| |8|-@19||+1#0000000#ffffff0|~+0#4040ff13&| @35"
command line..script /tmp/cirrus-ci-build/src/testdir/runtest.vim[677]..function RunTheTest[63]..Test_diff_screen[40]..VerifyBoth[15]..VerifyScreenDump line 102: unified: See dump file difference: call term_dumpdiff("testdir/failed/Test_diff_02.dump", "testdir/dumps/Test_diff_02.dump"); difference in line 1: "| +0#0000e05#a8a8a8255@1|0+0#0000000#5fd7ff255| @33||+1&#ffffff0| +0#0000e05#a8a8a8255@1|1+0#0000000#ffffff0| @33"; difference in line 2: "| +0#0000e05#a8a8a8255@1|1+0#0000000#ffffff0| @33||+1&&| +0#0000e05#a8a8a8255@1|-+0#4040ff13#afffff255@34"; difference in line 8: "|++0#0000e05#a8a8a8255| |+|-@1| @1|4| |l|i|n|e|s|:| |7|-@19||+1#0000000#ffffff0| +0#0000e05#a8a8a8255@1|7+0#0000000#ffffff0| @33"; difference in line 9: "|~+0#4040ff13&| @35||+1#0000000&|++0#0000e05#a8a8a8255| |+|-@1| @1|3| |l|i|n|e|s|:| |8|-@19"; difference in line 10: "|~+0#4040ff13#ffffff0| @35||+1#0000000&|~+0#4040ff13&| @35"
Run 3, 15:08:11 - 15:08:15 in  3.456647 seconds:
command line..script /tmp/cirrus-ci-build/src/testdir/runtest.vim[677]..function RunTheTest[63]..Test_diff_screen[36]..VerifyBoth[15]..VerifyScreenDump line 102: unified: See dump file difference: call term_dumpdiff("testdir/failed/Test_diff_01.dump", "testdir/dumps/Test_diff_01.dump"); difference in line 1: "| +0#0000e05#a8a8a8255@1|1+0#0000000#ffffff0| @33||+1&&| +0#0000e05#a8a8a8255@1|0+0#0000000#5fd7ff255| @33"; difference in line 2: "| +0#0000e05#a8a8a8255@1|-+0#4040ff13#afffff255@34||+1#0000000#ffffff0| +0#0000e05#a8a8a8255@1|1+0#0000000#ffffff0| @33"; difference in line 8: "| +0#0000e05#a8a8a8255@1|7+0#0000000#ffffff0| @33||+1&&|++0#0000e05#a8a8a8255| |+|-@1| @1|4| |l|i|n|e|s|:| |7|-@19"; difference in line 9: "|+| |+|-@1| @1|3| |l|i|n|e|s|:| |8|-@19||+1#0000000#ffffff0|~+0#4040ff13&| @35"
command line..script /tmp/cirrus-ci-build/src/testdir/runtest.vim[677]..function RunTheTest[63]..Test_diff_screen[40]..VerifyBoth[15]..VerifyScreenDump line 102: unified: See dump file difference: call term_dumpdiff("testdir/failed/Test_diff_02.dump", "testdir/dumps/Test_diff_02.dump"); difference in line 1: "| +0#0000e05#a8a8a8255@1|0+0#0000000#5fd7ff255| @33||+1&#ffffff0| +0#0000e05#a8a8a8255@1|1+0#0000000#ffffff0| @33"; difference in line 2: "| +0#0000e05#a8a8a8255@1|1+0#0000000#ffffff0| @33||+1&&| +0#0000e05#a8a8a8255@1|-+0#4040ff13#afffff255@34"; difference in line 8: "|++0#0000e05#a8a8a8255| |+|-@1| @1|4| |l|i|n|e|s|:| |7|-@19||+1#0000000#ffffff0| +0#0000e05#a8a8a8255@1|7+0#0000000#ffffff0| @33"; difference in line 9: "|~+0#4040ff13&| @35||+1#0000000&|++0#0000e05#a8a8a8255| |+|-@1| @1|3| |l|i|n|e|s|:| |8|-@19"; difference in line 10: "|~+0#4040ff13#ffffff0| @35||+1#0000000&|~+0#4040ff13&| @35"
Run 4, 15:08:21 - 15:08:24 in  3.370193 seconds:
command line..script /tmp/cirrus-ci-build/src/testdir/runtest.vim[677]..function RunTheTest[63]..Test_diff_screen[36]..VerifyBoth[15]..VerifyScreenDump line 102: unified: See dump file difference: call term_dumpdiff("testdir/failed/Test_diff_01.dump", "testdir/dumps/Test_diff_01.dump"); difference in line 1: "| +0#0000e05#a8a8a8255@1|1+0#0000000#ffffff0| @33||+1&&| +0#0000e05#a8a8a8255@1|0+0#0000000#5fd7ff255| @33"; difference in line 2: "| +0#0000e05#a8a8a8255@1|-+0#4040ff13#afffff255@34||+1#0000000#ffffff0| +0#0000e05#a8a8a8255@1|1+0#0000000#ffffff0| @33"; difference in line 8: "| +0#0000e05#a8a8a8255@1|7+0#0000000#ffffff0| @33||+1&&|++0#0000e05#a8a8a8255| |+|-@1| @1|4| |l|i|n|e|s|:| |7|-@19"; difference in line 9: "|+| |+|-@1| @1|3| |l|i|n|e|s|:| |8|-@19||+1#0000000#ffffff0|~+0#4040ff13&| @35"
command line..script /tmp/cirrus-ci-build/src/testdir/runtest.vim[677]..function RunTheTest[63]..Test_diff_screen[40]..VerifyBoth[15]..VerifyScreenDump line 102: unified: See dump file difference: call term_dumpdiff("testdir/failed/Test_diff_02.dump", "testdir/dumps/Test_diff_02.dump"); difference in line 1: "| +0#0000e05#a8a8a8255@1|0+0#0000000#5fd7ff255| @33||+1&#ffffff0| +0#0000e05#a8a8a8255@1|1+0#0000000#ffffff0| @33"; difference in line 2: "| +0#0000e05#a8a8a8255@1|1+0#0000000#ffffff0| @33||+1&&| +0#0000e05#a8a8a8255@1|-+0#4040ff13#afffff255@34"; difference in line 8: "|++0#0000e05#a8a8a8255| |+|-@1| @1|4| |l|i|n|e|s|:| |7|-@19||+1#0000000#ffffff0| +0#0000e05#a8a8a8255@1|7+0#0000000#ffffff0| @33"; difference in line 9: "|~+0#4040ff13&| @35||+1#0000000&|++0#0000e05#a8a8a8255| |+|-@1| @1|3| |l|i|n|e|s|:| |8|-@19"; difference in line 10: "|~+0#4040ff13#ffffff0| @35||+1#0000000&|~+0#4040ff13&| @35"
Run 5, 15:08:32 - 15:08:36 in  3.485333 seconds:
command line..script /tmp/cirrus-ci-build/src/testdir/runtest.vim[677]..function RunTheTest[63]..Test_diff_screen[36]..VerifyBoth[15]..VerifyScreenDump line 102: unified: See dump file difference: call term_dumpdiff("testdir/failed/Test_diff_01.dump", "testdir/dumps/Test_diff_01.dump"); difference in line 1: "| +0#0000e05#a8a8a8255@1|1+0#0000000#ffffff0| @33||+1&&| +0#0000e05#a8a8a8255@1|0+0#0000000#5fd7ff255| @33"; difference in line 2: "| +0#0000e05#a8a8a8255@1|-+0#4040ff13#afffff255@34||+1#0000000#ffffff0| +0#0000e05#a8a8a8255@1|1+0#0000000#ffffff0| @33"; difference in line 8: "| +0#0000e05#a8a8a8255@1|7+0#0000000#ffffff0| @33||+1&&|++0#0000e05#a8a8a8255| |+|-@1| @1|4| |l|i|n|e|s|:| |7|-@19"; difference in line 9: "|+| |+|-@1| @1|3| |l|i|n|e|s|:| |8|-@19||+1#0000000#ffffff0|~+0#4040ff13&| @35"
command line..script /tmp/cirrus-ci-build/src/testdir/runtest.vim[677]..function RunTheTest[63]..Test_diff_screen[40]..VerifyBoth[15]..VerifyScreenDump line 102: unified: See dump file difference: call term_dumpdiff("testdir/failed/Test_diff_02.dump", "testdir/dumps/Test_diff_02.dump"); difference in line 1: "| +0#0000e05#a8a8a8255@1|0+0#0000000#5fd7ff255| @33||+1&#ffffff0| +0#0000e05#a8a8a8255@1|1+0#0000000#ffffff0| @33"; difference in line 2: "| +0#0000e05#a8a8a8255@1|1+0#0000000#ffffff0| @33||+1&&| +0#0000e05#a8a8a8255@1|-+0#4040ff13#afffff255@34"; difference in line 8: "|++0#0000e05#a8a8a8255| |+|-@1| @1|4| |l|i|n|e|s|:| |7|-@19||+1#0000000#ffffff0| +0#0000e05#a8a8a8255@1|7+0#0000000#ffffff0| @33"; difference in line 9: "|~+0#4040ff13&| @35||+1#0000000&|++0#0000e05#a8a8a8255| |+|-@1| @1|3| |l|i|n|e|s|:| |8|-@19"; difference in line 10: "|~+0#4040ff13#ffffff0| @35||+1#0000000&|~+0#4040ff13&| @35"
Flaky test failed too often, giving up
*** Error code 1```

but its still not useful as it fix not the error

pheiduck committed:
Modify build and test scripts in .cirrus.yml

Removed vterm option from configure command and adjusted term_wait in test script.

pheiduck committed:
Fix path in sed command for test script

pheiduck committed:
Fix sed command syntax in test script
